### PR TITLE
B-19463 add assigned user col to SC queue; add dropdown select to row

### PR DESCRIPTION
**Pull request by paulstonebraker**
## [INT PR](https://github.com/transcom/mymove/pull/13581)

> [!IMPORTANT]
> changes differences in this PR vs PR to INT are in ServicesCounselingQueue.jsx, I believe as a result of @JonSpight's work for counseling offices hitting INT after my PR, and merging to main before this PR... as always, your extra eyes are appreciated :) 

This is initial queue management front end work for the Services Counselors Queue, adding an assigned column, along with a dropdown. The dropdown currently does not handle assign/un-assign functionality (that'll be dealt with in future story [20529](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20529)

Current functionality is as follows:
-Supervisor's dropdown is populated first with a "-" to be used later for unassigning, then with all Services Counselors working at the logged in Supervisor's transportation office, and pushed onto the end, whoever it is assigned to (if it is assigned to someone outside of the office users available in the dropdown)
-Non supervisor's dropdown is populated first with a "-", then with the office user themself (for self assigning purposes), then pushed onto the end, whoever it is assigned to (if it is assigned to someone besides themself)

### CODE REVIEW

**Review by joeydoyecaci — CHANGES_REQUESTED** [review]
Changes were made on ghc.yaml in swagger & swagger-def, adding AssignedOfficeUser and assignedTo respectively in this PR, but not in the int PR. These changes must be removed.

**Comment by paulstonebraker** [comment]
> Changes were made on ghc.yaml in swagger & swagger-def, adding AssignedOfficeUser and assignedTo respectively in this PR, but not in the int PR. These changes must be removed.

My bad! I mistakenly opened this review and then took a half day and didn't undo my mistake 🤦  all should be good now

File: src/pages/Office/ServicesCounselingQueue/ServicesCounselingQueue.jsx [hunk 1]
```diff
@@ -48,147 +48,171 @@ import ConnectedFlashMessage from 'containers/FlashMessage/FlashMessage';
...
-          Filter: (props) => (
-            <MultiSelectTypeAheadCheckBoxFilter
-              options={originLocationList}
-              placeholder="Start typing a duty location..."
-              // eslint-disable-next-line react/jsx-props-no-spreading
-              {...props}
-            />
-          ),
+        }),
+    createHeader('Counseling office', 'counselingOffice', {
```
Comment (pambecker) [inline src/pages/Office/ServicesCounselingQueue/ServicesCounselingQueue.jsx:189]: The line count diff here seems to deal with Counseling office column being in main already vs it was not when you merged this code to INT.  Still seems good to go here, just moved lines from 187-191 up.

**Review by pambecker — APPROVED** [review]
Seems to be good.  as noted, differences seem to be due to Counseling office in main and it wasn't yet in int when that merged occurred.  built the branch locally and the Assigned column seems to function as expected.

**Review by joeydoyecaci — APPROVED** [review]

**Pull request merged**